pkg/s3wrapper/client.go

```diff
@@ -161,6 +168,200 @@ func (c *S3Client) UploadFile(ctx context.Context, filePath, objectName string)
+	getRest, err := c.client.GetObject(&s3.GetObjectInput{
+		Bucket: aws.String(c.cfg.S3Bucket),
+		Key:    aws.String(baseObjectName),
+		Range:  aws.String(fmt.Sprintf("bytes=%d-%d", areaOffsetBytes, areaOffsetBytes+fiveMB-1)),
+	})
+	if err != nil {
+		err = errors.Wrapf(err, "Failed to fetch embedded area of live ISO %s", objectName)
+		log.Error(err)
+		return err
+	}
+	origContents, err := ioutil.ReadAll(getRest.Body)
+	if err != nil {
+		err = errors.Wrapf(err, "Failed to fetch body from embedded area of live ISO %s", objectName)
+		log.Error(err)
+		return err
+	}
+
+	completedPart, err = c.uploadIgnition(log, aws.Int64(2), uploadID, objectName, ignitionConfig, origContents, areaLengthBytes)
+	if err != nil {
+		return err
+	}
+	completedParts = append(completedParts, completedPart)
+
+	// Third part: copy the last part of the live ISO, after the embedded area
+	completedPart, err = c.uploadPartCopy(log, aws.Int64(3), uploadID, baseObjectName, objectName, areaOffsetBytes+fiveMB, baseObjectSize-1)
+	if err != nil {
+		return err
+	}
+	completedParts = append(completedParts, completedPart)
+
+	_, err = c.client.CompleteMultipartUpload(&s3.CompleteMultipartUploadInput{
+		Bucket:   aws.String(c.cfg.S3Bucket),
+		Key:      aws.String(objectName),
+		UploadId: uploadID,
+		MultipartUpload: &s3.CompletedMultipartUpload{
+			Parts: completedParts,
+		},
+	})
+	if err != nil {
+		err = errors.Wrapf(err, "Failed to complete upload for %s", objectName)
+		log.Error(err)
+		return err
+	}
+	return nil
+}
+
+func (c *S3Client) getISOHeaderInfo(log logrus.FieldLogger, baseObjectName string) (int64, int64, error) {
+	// Download header of the live ISO (last 24 bytes of the first 32KB)
+	getResp, err := c.client.GetObject(&s3.GetObjectInput{
+		Bucket: aws.String(c.cfg.S3Bucket),
+		Key:    aws.String(baseObjectName),
+		Range:  aws.String("bytes=32744-32768"),
+	})
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get header of object %s from bucket %s", baseObjectName, c.cfg.S3Bucket)
+		return 0, 0, err
+	}
+	headerString, err := ioutil.ReadAll(getResp.Body)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to read header of object %s from bucket %s", baseObjectName, c.cfg.S3Bucket)
+		return 0, 0, err
+	}
+
+	res := bytes.Compare(headerString[0:8], []byte("coreiso+"))
+	if res != 0 {
+		return 0, 0, errors.New("Could not find magic string in object header")
+	}
+
+	offset := int64(binary.LittleEndian.Uint64(headerString[8:16]))
+	length := int64(binary.LittleEndian.Uint64(headerString[16:24]))
+
+	// For now we assume that the embedded area is less than 5MB, which is the minimum S3 part size
+	if length > int64(fiveMB) {
+		return 0, 0, errors.New("ISO embedded area larger than what is currently supported")
+	}
+	return offset, length, nil
+}
+
+func (c *S3Client) uploadPartCopy(log logrus.FieldLogger, partNum *int64, uploadID *string, sourceObjectKey string, destObjectKey string,
```

Comment:
Shouldn't be a pointer. You print and use the address of the pointer in this function.